upgrade/1.4.4/README.md

```diff
@@ -0,0 +1,392 @@
+INFO: Summary of number of components in each status: configured: 9
+====> Completed update of CFS configuration(s)
+====> Cleaning up install dependencies
+```
+
+When configuration of all components is successful, the summary line will show all components
+with status "configured".
+
+### Update NCN images
+
+NCN images must be rebuilt at this time in order to acquire an important Kernel panic mitigation. The mitigation entails
+a new Kernel and networking drivers for SP4 images \(Kubernetes\), as well as the blacklisting of the QLogic RDMA driver
+for SP4 and SP3 (all NCNs).
+
+Despite rebuilding both Kubernetes and Storage CEPH images, **only Kubernetes nodes will embark on a rolling rebuilt**.
+Storage CEPH nodes will receive an in-place modification, and do not need to be rebuilt at this time.
+
+#### Warnings
+
+***IMPORTANT*** This minor version bump has an unprecedented rolling rebuild. This is a friendly reminder that any 
+system administration data living on masters and workers **will be wiped** during the rebuild. Administrators are
+advised to take backups of their local, site files.
+
+> Examples:
+>
+> * ` ~/.config/sat/sat`
+> * `/etc/motd`
+> * `/etc/sudoers`
+> * `/home`
+> * `/root/.ssh/config`
+
+
+#### Image customization
+
+1. Follow option one of the following options based on your use case:
+
+   * [Option 2: Upgrade of CSM on system with additional products](../Stage_0_Prerequisites.md#option-2-upgrade-of-csm-on-system-with-additional-products)
+   * [Option 3: Upgrade of CSM on CSM-only system](../Stage_0_Prerequisites.md#option-3-upgrade-of-csm-on-csm-only-system)
+
+   > ***NOTE*** If [Option 1: Upgrade of CSM and additional products](../Stage_0_Prerequisites.md#option-1-upgrade-of-csm-and-additional-products) is better suiting, then
+   > do a full IUF run upgrading all products including CSM.
+   
+1. Follow the directions in [Stage 0.4](../Stage_0_Prerequisites.md#stage-04---backup-workload-manager-data).
+
+### Storage nodes in-place update
+
+In lieu of rebuilding the storage nodes, they will be live patched.
+
+1. (`ncn-m001#`) Unload and blacklist the QLogic RDMA `qedr` driver.
+
+   ```bash
+   /usr/share/doc/csm/upgrade/1.4.4/scripts/storage-in-place-patch.sh
+   ```
+
+1. (`ncn-m001#`) Verify that `qedr` is no longer loaded.
+
+   ```bash
+   pdsh -b -w $(grep -oP 'ncn-s\d+' /etc/hosts | sort -u | tr -t '\n' ',') '
+   lsmod | grep -Eo '\''^qedr'\'' || echo OK
+   ' | dshbak -c
+   ```
+
+   Expected output:
+
+   ```text
+   ----------------
+   ncn-s[001-003]
+   ----------------
+   OK
+   ```
+
+### Kubernetes nodes rolling rebuild
+
+1. (`ncn-m001#`) Set environment variables
+
+   ```bash
+   export CSM_REL_NAME="csm-${CSM_RELEASE}"
+   export CSM_ARTI_DIR="/etc/cray/upgrade/csm/${CSM_REL_NAME}/tarball/${CSM_REL_NAME}"
+   ```
+
+1. Proceed with the following sections from Stage 1:
+
+   * [Start typescript on `ncn-m001`](../Stage_1.md#start-typescript-on-ncn-m001)
```

Comment:
If option 3 was followed, I believe the CSM images would not be set correctly on the NCNs. It is usually prerequisites that sets the CSM images (unless IUF is being used in which case the images are re-set to the images created by IUF). But in this case, prerequisites is not bing run so for option 3, the images are not set by this point in time.

Reply:
I see, Option two has step 12 which points users to the necessary steps for updating BSS. Option three does not have such a step.

Reply:
Studying option 3 more, as it's doing a lot less than option 2.

Reply:
I think if someone is doing a CSM only upgrade, they need to be directed to follow step 2. However, there would not have been an IUF run so we would need to create additional instructions on how to find/create the right bootprep files.